src/app/device-model/device-model-edit/device-model-edit.component.ts

```diff
@@ -31,7 +31,7 @@ export class DeviceModelEditComponent implements OnInit {
   controlledPropperties = [];
   categories = [];
   supportedUnits = new SupportedUnit();
-  deviceFunctions = [];
+  deviceFunctions: string[] = [];
```

Comment:
Nice that you added types! Technically, it is of type `DeviceFunction`. You won't get any TS errors if someone adds a non-string device function value in the future.